src/nvlink/wrapper_32bit.py

```
@@ -0,0 +1,583 @@
+            raise NVLinkError("NV-Link stream not open. Call nv_open() or nv_rt_open() first.")
+
+        try:
+            # 32-bit Python: Direct call with proper parameter types
+            nv_result = self._nvlink.NVRead("", BUFFER_SIZE_NVREAD, "")
+
+            if isinstance(nv_result, tuple) and len(nv_result) >= 4:
+                result = nv_result[0]
+                buff_str = nv_result[1]
+                filename_str = nv_result[3]
+            else:
+                raise NVLinkError(f"Unexpected NVRead return format: {type(nv_result)}")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)